tooling/minitest/src/pass/lock_handover.rs

```diff
@@ -0,0 +1,55 @@
+use crate::*;
+
+#[test]
+/// This test is written in response to a bug that was found.
+/// The bug was: When handing over a lock instead of making the lock owned by the acquiring thread
+/// it was once again owned by the releasing thread.
+/// This lead to UB once the acquirer tried to release the lock.
+/// 
+/// What it wants to check is: Does the lock handover work correctly?
+/// By making the critical section extremly large we get a high probability that the hand over happend.
```

Comment:
"extremely"? :) If I see correctly it is counting to 256, right?

Reply:
Maybe 256 isn't that big... but I'd claim $2^{-256}$ is extremely small. :)